packages/node-resolve/src/package/resolvePackageImportsExports.ts

```diff
@@ -0,0 +1,98 @@
+/* eslint-disable no-await-in-loop */
+import resolvePackageTarget from './resolvePackageTarget';
+
+import { InvalidModuleSpecifierError } from './utils';
+
+/**
+ * Implementation of Node's `PATTERN_KEY_COMPARE` function
+ */
+function nodePatternKeyCompare(keyA: string, keyB: string) {
+  // Let baseLengthA be the index of "*" in keyA plus one, if keyA contains "*", or the length of keyA otherwise.
+  const baseLengthA = keyA.includes('*') ? keyA.indexOf('*') + 1 : keyA.length;
+  // Let baseLengthB be the index of "*" in keyB plus one, if keyB contains "*", or the length of keyB otherwise.
+  const baseLengthB = keyB.includes('*') ? keyB.indexOf('*') + 1 : keyB.length;
+
+  // if baseLengthA is greater, return -1, if lower 1
+  const rval = baseLengthB - baseLengthA;
+  if (rval !== 0) return rval;
+
+  // If keyA does not contain "*", return 1.
+  if (!keyA.includes('*')) return 1;
+  // If keyB does not contain "*", return -1.
+  if (!keyB.includes('*')) return -1;
+
+  // If the length of keyA is greater than the length of keyB, return -1.
+  // If the length of keyB is greater than the length of keyA, return 1.
+  // --> both keys do not contain * so baseLength of both is the key length, so this is already handled by `rval` above
+
+  // Else Return 0.
+  return 0;
+}
+
+interface ParamObject {
+  matchKey: string;
+  matchObj: Record<string, string>;
+  isImports?: boolean;
+}
+async function resolvePackageImportsExports(
+  context: any,
+  { matchKey, matchObj, isImports }: ParamObject
+) {
+  // If matchKey is a key of matchObj and does not contain "*", then
+  if (!matchKey.includes('*') && matchKey in matchObj) {
+    // Let target be the value of matchObj[matchKey].
+    const target = matchObj[matchKey];
+    // Return the result of PACKAGE_TARGET_RESOLVE(packageURL, target, null, isImports, conditions).
+    const resolved = await resolvePackageTarget(context, { target, patternMatch: '', isImports });
+    return resolved;
+  }
+
+  // Let expansionKeys be the list of keys of matchObj containing only a single "*"
+  const expansionKeys = Object.keys(matchObj)
+    // Assert: ends with "/" or contains only a single "*".
+    .filter((k) => k.endsWith('/') || k.includes('*'))
```

Comment:
This is how the algorithm is written - basically multiple `**` are invalid, but we don't ban them outright in case it might be supported in future - so instead they are filtered out by omission since they can never be matched.